prometheus_hardware_exporter/collectors/perccli.py

```diff
@@ -0,0 +1,132 @@
+"""Collector for PowerEdgeRAID controller."""
+import socket
+from logging import getLogger
+from typing import Any, Dict, List
+
+from ..utils import Command, Result, get_json_output
+
+logger = getLogger(__name__)
+
+
+class PercCLI(Command):
+    """Command line tool for PowerEdge RAID Controller."""
+
+    prefix = "sudo"
+    command = "/opt/MegaRAID/perccli/perccli64"
+
+    def _get_controllers(self) -> Result:
+        result = self("/call show j")
+        if result.error:
+            logger.error(result.error)
+            return result
+
+        assert isinstance(result.data, str)
+        result = get_json_output(result.data)
+        if result.error:
+            return result
+        return result
+
+    def _cmd_status(self, controller: Dict[str, Any]) -> bool:
+        if controller["Command Status"]["Status"] == "Success":
+            return True
+        return False
+
+    def _ctrl_exists(self, controllers: Dict[str, Any]) -> bool:
+        if (
+            controllers["Controllers"][0]["Command Status"]["Status"] == "Failure"
```

Comment:
nitpick: It's not quite readable. Maybe consider creating variables for status and command status?

Reply:
Creating a variable for this kind of check is not that worthy. But I can add an example output json to the doc-string for better understanding.
How do you think?

Reply:
Sorry for not being clear, what I mean is, for example `status = controllers["Controllers"][0]["Command Status"]["Status"]`, then `if status == "Failure"`. It take 3 lines to read the `if` condition

Reply:
```suggestion
            status = controllers["Controllers"][0]["Command Status"]["Status"]
            description = controllers["Controllers"][0]["Command Status"]["Description"]
            return not (status == "Failure" and description == "No Controller found")
```